Blish HUD/GameServices/ModuleService.cs

```diff
@@ -317,20 +320,61 @@ public class ModuleService : GameService {
         }
 
         private void Gw2Instance_Gw2Started(object sender, EventArgs e) {
-            foreach (var module in _modules) {
-                if (module.State.Enabled) {
-                    module.TryEnable();
+            var resolvedDependencyOrder = SortByDependency(_modules, module => {
+                var dependencyModules = _modules.Where(m => {
+                    // Get all modules which have a dependency on the current module
+                    return module.Manifest?.Dependencies?.Any(dm => !dm.IsBlishHud && dm.Namespace == m.Manifest.Namespace) ?? false;
+                });
+
+                return dependencyModules;
+            }).ToList();
+
+            if (resolvedDependencyOrder != null) {
+                Logger.Debug($"Resolved dependency order: {string.Join(", ", resolvedDependencyOrder.Select(x => x.Manifest.Namespace).ToArray())}");
+
+                foreach (var module in resolvedDependencyOrder) {
+                    if (module.State.Enabled) {
+                        module.TryEnable();
+                    }
+                }
+            }else {
+                Logger.Error($"Could not resolve dependencies of any module.");
```

Comment:
If this is truly an unexpected error, we should provide more information about the error.  Otherwise, the entry in Sentry will likely be useless.

Reply:
I think I can remove that if again. I only added that because SonarCloud complained I should add it. After I added it, it complained that it is useless..

Reply:
SonarCloud is truly hilarious sometimes.  :D